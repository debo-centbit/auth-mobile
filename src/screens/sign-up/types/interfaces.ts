import { ReactNode } from "react";
import { Position } from "./enums";

export interface UserDetailsProps {
  firstName?: string;
  lastName?: string;
  fullName?: string;
  email?: string;
  phoneNumber?: string;
  password?: string;
  [key: string]: any;
}

export interface StyleProps {
  color: string;
  top: number;
  left: number;
  position: Position;
  borderRadius: number;
  bgColor: string;
  width: number;
  height: number;
  inputBorderRadius: number;
  inputMarginTop: number;
  backBtnLeft: number;
}

export interface ButtonProps {
  borderRadius: number;
  children: ReactNode | string;
  top: number;
  left: number;
  position: Position;
  color: string;
  bgColor: string;
  btnWidth: number;
  btnHeight: number;
  testID: string;
  onPress: () => void;
}

export interface FieldConfig {
  name: string;
  placeholder: string;
}

export interface FormFieldsProps {
  fieldsConfig: FieldConfig[];
  handleChange: (fieldName: string, value: string) => void;
  userDetails: UserDetailsProps;
  borderRadius: number;
  marginTop: number;
}

export interface LogoProps {
  imageUrl: string;
  logoWidth: number;
  logoHeight: number;
  logoPosition: Position;
  logoTop: number;
  logoLeft: number;
  containerTestID: string;
  imageTestID: string;
}
