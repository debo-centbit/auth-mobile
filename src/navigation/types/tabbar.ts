import { RouteNames } from "../route-names";

// Bottom tab bar

export type BottomTabParamList = {
  [RouteNames.homeStack]: undefined;
  [RouteNames.profileStack]: undefined;
  [RouteNames.passwordResetStack]: undefined;

};
