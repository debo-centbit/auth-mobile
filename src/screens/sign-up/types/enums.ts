export enum Position {
    ABSOLUTE = "absolute",
    RELATIVE = "relative",
  }
  