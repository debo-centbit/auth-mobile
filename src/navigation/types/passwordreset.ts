import { BottomTabNavigationProp } from "@react-navigation/bottom-tabs";
import { CompositeNavigationProp, RouteProp } from "@react-navigation/native";
import { StackNavigationProp, StackScreenProps } from "@react-navigation/stack";
import { RouteNames } from "../route-names"; 
import { BottomTabParamList } from "./tabbar"; 


export type PasswordResetStackParamList= {
    [RouteNames.passwordReset]: undefined;
  };

export type PasswordResetStackNavigationProp<
  RouteName extends keyof PasswordResetStackParamList
> = CompositeNavigationProp<
  StackNavigationProp<PasswordResetStackParamList, RouteName>,
  BottomTabNavigationProp<BottomTabParamList>
>;

export type PasswordResetStackRouteProp<
  RouteName extends keyof PasswordResetStackParamList
> = RouteProp<PasswordResetStackParamList, RouteName>;

export type PasswordResetStackScreenProp<
  RouteName extends keyof PasswordResetStackParamList
> = StackScreenProps<PasswordResetStackParamList, RouteName>;
