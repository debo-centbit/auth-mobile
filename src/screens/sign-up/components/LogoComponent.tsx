import React from "react";
import { Image, View, StyleSheet } from "react-native";
import { LogoProps } from "../types/interfaces";

const LogoComponent: React.FC<LogoProps> = ({
  imageUrl,
  logoHeight,
  logoWidth,
  logoPosition,
  logoTop,
  logoLeft,
  containerTestID,
  imageTestID,
}) => {
  const styles = StyleSheet.create({
    logoContainer: {
      width: "fit-content",
      height: "fit-content",
      position: logoPosition,
      top: logoTop,
      left: logoLeft,
    },
    logo: {
      width: logoWidth,
      height: logoHeight,
    },
  });

  return (
    <View style={styles.logoContainer} testID={containerTestID}>
      <Image
        source={{ uri: imageUrl }}
        style={styles.logo}
        testID={imageTestID}
      />
    </View>
  );
};

export default LogoComponent;
