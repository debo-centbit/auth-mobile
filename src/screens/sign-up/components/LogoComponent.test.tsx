import React from "react";
import { render } from "@testing-library/react-native";
import LogoComponent from "./LogoComponent";
import * as Jest from "@testing-library/jest-native/extend-expect";
import { Position } from "../types/enums";

describe("LogoComponent", () => {
  const mockProps = {
    imageUrl: "https://example.com/logo.png",
    logoWidth: 100,
    logoHeight: 50,
    logoPosition: Position.ABSOLUTE,
    logoTop: 20,
    logoLeft: 10,
    containerTestID: "logo-container",
    imageTestID: "logo-image",
  };

  it("renders correctly with provided props", () => {
    const { getByTestId } = render(<LogoComponent {...mockProps} />);
    const logoContainer = getByTestId(mockProps.containerTestID);
    const logoImage = getByTestId(mockProps.imageTestID);

    expect(logoContainer).toBeDefined();
    expect(logoImage).toBeDefined();

    // Check styles
    expect(logoContainer).toHaveStyle({
      position: mockProps.logoPosition,
      top: mockProps.logoTop,
      left: mockProps.logoLeft,
    });
    expect(logoImage).toHaveStyle({
      width: mockProps.logoWidth,
      height: mockProps.logoHeight,
    });
  });

  it("matches the snapshot", () => {
    const { toJSON } = render(<LogoComponent {...mockProps} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
