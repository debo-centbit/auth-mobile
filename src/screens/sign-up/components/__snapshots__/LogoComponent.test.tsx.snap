// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoComponent matches the snapshot 1`] = `
<View
  style={
    Object {
      "height": "fit-content",
      "left": 10,
      "position": "absolute",
      "top": 20,
      "width": "fit-content",
    }
  }
  testID="logo-container"
>
  <Image
    source={
      Object {
        "uri": "https://example.com/logo.png",
      }
    }
    style={
      Object {
        "height": 50,
        "width": 100,
      }
    }
    testID="logo-image"
  />
</View>
`;
